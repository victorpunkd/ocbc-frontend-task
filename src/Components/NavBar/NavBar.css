.navBarContainer {
  width: 100%;
  text-align: right;
  padding: 5% 0%;
}

.logoutButton {
  font-weight: 700;
  cursor: pointer;
}

.logoutButton:hover {
  color: gray;
}
