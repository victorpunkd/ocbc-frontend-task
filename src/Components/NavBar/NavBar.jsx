import React from "react";
import { useNavigate } from "react-router-dom";
import "./NavBar.css";

const NavBar = () => {
  const navigate = useNavigate();
  const handleLogoutClick = () => {
    sessionStorage.removeItem("token");
    navigate("/");
  };
  return (
    <div className="navBarContainer">
      <div className="logoutButton" onClick={handleLogoutClick}>
        Logout
      </div>
    </div>
  );
};

export default NavBar;
