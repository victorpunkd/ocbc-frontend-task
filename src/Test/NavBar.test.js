import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import NavBar from "../Components/NavBar/NavBar";

configure({ adapter: new Adapter() });

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

describe("NavBar", () => {
  it("snapshot renders", () => {
    const component = renderer.create(<NavBar />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render logoutButton div", () => {
    const wrapper = shallow(<NavBar />);
    expect(wrapper.find(".logoutButton").exists()).toBe(true);
  });

  it("Should simulate logout onClick event", () => {
    const wrapper = shallow(<NavBar />);
    wrapper.find(".logoutButton").simulate("click");

    expect(mockedUsedNavigate).toHaveBeenCalled();
  });
});
