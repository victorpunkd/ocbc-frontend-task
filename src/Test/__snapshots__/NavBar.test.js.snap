// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar snapshot renders 1`] = `
<div
  className="navBarContainer"
>
  <div
    className="logoutButton"
    onClick={[Function]}
  >
    Logout
  </div>
</div>
`;
